Todo/TaskListsViewController.swift

```diff
@@ -0,0 +1,271 @@
+    var database: CBLDatabase!
+    
+    var listsLiveQuery: CBLLiveQuery!
+    var listRows : [CBLQueryRow]?
+    
+    var incompTasksCountsLiveQuery: CBLLiveQuery!
+    var incompTasksCounts : [String : Int]?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Setup Navigation bar:
+        if !kLoginFlowEnabled {
+            // Remove logout button:
+            self.navigationItem.leftBarButtonItem = nil
+        }
+        
+        // Setup SearchController:
+        searchController = UISearchController(searchResultsController: nil)
+        searchController.searchResultsUpdater = self
+        self.tableView.tableHeaderView = searchController.searchBar
+        
+        // Get database and username:
+        let app = UIApplication.sharedApplication().delegate as! AppDelegate
+        database = app.database
+        username = Session.username
+        
+        // Setup view and query:
+        setupViewAndQuery()
+    }
+
+    deinit {
+        if listsLiveQuery != nil {
+            listsLiveQuery.removeObserver(self, forKeyPath: "rows")
+            listsLiveQuery.stop()
+        }
+        
+        if incompTasksCountsLiveQuery != nil {
+            incompTasksCountsLiveQuery.removeObserver(self, forKeyPath: "rows")
+            incompTasksCountsLiveQuery.stop()
+        }
+    }
+    
+    // MARK: - Navigation
+    
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        if let tabBarController = segue.destinationViewController as? UITabBarController {
+            let selectedList = listRows![self.tableView.indexPathForSelectedRow!.row].document
+            
+            let tasksController = tabBarController.viewControllers![0] as! TasksViewController
+            tasksController.taskList = selectedList
+            
+            let usersController = tabBarController.viewControllers![1] as! UsersViewController
+            usersController.taskList = selectedList
+        }
+    }
+    
+    // MARK: - KVO
+    
+    override func observeValueForKeyPath(keyPath: String?, ofObject object: AnyObject?,
+        change: [String : AnyObject]?, context: UnsafeMutablePointer<Void>) {
+            if object as? NSObject == listsLiveQuery {
+                reloadTaskLists(listsLiveQuery.rows)
+            } else if object as? NSObject == incompTasksCountsLiveQuery {
+                reloadIncompleteTasksCounts(incompTasksCountsLiveQuery.rows)
+            }
+    }
+
+    // MARK: - UITableViewController
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return listRows?.count ?? 0
+    }
+
+    override func tableView(tableView: UITableView,
+        cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+            let cell = tableView.dequeueReusableCellWithIdentifier("TaskListCell") as UITableViewCell!
+            
+            let doc = listRows![indexPath.row].document!
```

Comment:
It's faster if you don't have to load the document for every row. Instead, get the name as `row.key` and the docID as `row.documentID`.